tree/ntuple/v7/test/ntuple_type_name.cxx

```diff
@@ -281,3 +281,26 @@ TEST(RNTuple, ContextDependentTypeNames)
       }
    }
 }
+
+TEST(RNTuple, RenormalizedDemangledName)
+{
+   // Test two cases for type name renormalization:
+   // - class with template argument that is normalized by the RNTuple IO (e.g. int to std::int32_t)
+   // - class with template argument which is another template class (e.g. myclass<std::vector<int>>) which by default
+   //   is stripped of the `std::` prefix and has a trailing blank space.
+   FileRaii fileGuard("test_ntuple_atlas_like_datavector.root");
+
+   {
+      auto model = RNTupleModel::Create();
+      auto f1 = model->MakeField<DataVector<int>>("f1");
+      auto f2 = model->MakeField<DataVector<int, std::vector<CustomStruct>>>("f2");
```

Comment:
Do we need to add 'new' dictionaries request to this test in order to fully test the user case?

Reply:
I have changed the second field to use `DataVector<bool, std::vector<unsigned int>>` which already has a dictionary request at https://github.com/root-project/root/blob/cb10c61bef3546cfbd24d12f4288eeb2c3655fd0/tree/ntuple/v7/test/CustomStructLinkDef.h#L30 and for the first field `DataVector<int>` has already a dictionary request at https://github.com/root-project/root/blob/cb10c61bef3546cfbd24d12f4288eeb2c3655fd0/tree/ntuple/v7/test/CustomStructLinkDef.h#L28 IIUC since the second template argument is defaulted to `double`.